logstash-core/lib/logstash/shutdown_controller.rb

```diff
@@ -47,10 +47,10 @@ def start
       cycle_number = 0
       stalled_count = 0
       Stud.interval(@cycle_period) do
-        @reports << Report.from_pipeline(@pipeline)
+        @reports << pipeline_report_snapshot
         @reports.delete_at(0) if @reports.size > @report_every # expire old report
         if cycle_number == (@report_every - 1) # it's report time!
-          logger.warn(@reports.last.to_hash)
+          logger.warn(@reports.last.to_simple_hash)
```

Comment:
it depends on the behavior you want.. I have been using to_str in another project for some stuff. 
`to_str` tells the ruby language the the instance you are passing has a string like api.

```ruby
class Message
  def initialize(title)
    @title = title
  end

  def to_str
    @title
  end
 alias_method :to_s, :to_str
end

m = Message.new("world")

puts "Hello  #{m}"
puts "Hello " + m
```
Defining both method allow both of this workflow to work, but in that case `to_s` would be fine.